include/swift/Runtime/Metadata.h

```diff
@@ -2643,23 +2649,47 @@ swift_getFunctionTypeMetadata(const void *flagsArgsAndResult[]);
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata1(FunctionTypeFlags flags,
-                               const void *arg0,
+                               const Metadata *arg0,
                                const Metadata *resultMetadata);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
+swift_getFunctionTypeMetadata1WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const uint32_t *paramFlags,
+                                        const Metadata *resultMetadata);
+
+SWIFT_RUNTIME_EXPORT
+const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata2(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
+                               const Metadata *arg0,
+                               const Metadata *arg1,
                                const Metadata *resultMetadata);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
-swift_getFunctionTypeMetadata3(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
-                               const void *arg2,
-                               const Metadata *resultMetadata);
+swift_getFunctionTypeMetadata2WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const Metadata *arg1,
+                                        const uint32_t *paramFlags,
+                                        const Metadata *resultMetadata);
+
+SWIFT_RUNTIME_EXPORT
+const FunctionTypeMetadata *swift_getFunctionTypeMetadata3(
+                                                FunctionTypeFlags flags,
+                                                const Metadata *arg0,
+                                                const Metadata *arg1,
+                                                const Metadata *arg2,
+                                                const Metadata *resultMetadata);
+
+SWIFT_RUNTIME_EXPORT
+const FunctionTypeMetadata *swift_getFunctionTypeMetadata3WithFlags(
+                                                FunctionTypeFlags flags,
+                                                const Metadata *arg0,
+                                                const Metadata *arg1,
+                                                const Metadata *arg2,
+                                                const uint32_t *paramFlags,
+                                                const Metadata *resultMetadata);
```

Comment:
Are you planning to collect that histogram of actual uses and then figure out what flags we really want in a follow-up?

Regardless, you should follow the same pattern here and take three uint32_t's.  Only the most general entrypoint should take it indirectly.